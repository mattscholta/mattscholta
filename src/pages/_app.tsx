import * as React from 'react';
import { CookiesProvider } from 'react-cookie';
import type { AppContext, AppProps } from 'next/app';
import App from 'next/app';
import parser from 'ua-parser-js';
import Cookies from 'universal-cookie';

import { intro } from '@src/config/intro';
import { IS_BROWSER } from '@src/config/settings';
import { AppComponent } from '@src/services/app/components/AppComponent';

// import { checkBasicAuth } from '@src/utils/auth';
import '@src/styles/index.css';

interface ApplicationProps extends AppProps {
  cookies: Record<string, any>;
  global: any;
}

/**
 * @name Application
 * @external https://nextjs.org/docs/advanced-features/custom-app
 * @description Next.js uses the App component to initialize pages.
 * You can override it and control the page initialization.
 */
const Application: React.FC<ApplicationProps> = (props, _context) => {
  const { cookies, global, Component, pageProps } = props;

  // Setup
  const cookieData = new Cookies(cookies);
  const cookieProps = IS_BROWSER ? {} : { cookies: cookieData };

  // Life Cycle
  React.useEffect(() => {
    intro();
  }, []);

  return (
    <CookiesProvider {...cookieProps}>
      <AppComponent global={global} pageProps={pageProps}>
        <Component {...pageProps} />
      </AppComponent>
    </CookiesProvider>
  );
};

/**
 * Only uncomment this method if you have blocking data requirements for
 * every single page in your application. This disables the ability to
 * perform automatic static optimization, causing every page in your app to
 * be server-side rendered.
 *
 * We connect to our GraphQL instance, potentially, on any page. This means
 * that any of our `useQuery` or `useMutation` actions will execute on the
 * server side, so we have to make our cookies available to the server
 */
(Application as any).getInitialProps = async (appContext: AppContext) => {
  const { ctx } = appContext;
  const { req, res: _res } = ctx;

  // Setup
  const cookies = IS_BROWSER ? document.cookie : (req as any)?.cookies ?? {};
  const global = {};
  const userAgent = req?.headers['user-agent'] ?? '';
  const ua = parser(userAgent);

  // Check auth will call res.end if password not matching.
  // if (req && res) checkBasicAuth(req, res);

  // Calls page's `getInitialProps` and fills `appProps.pageProps`
  const appProps = await App.getInitialProps(appContext);

  return { ...appProps, cookies, global, ua };
};

export { Application, Application as default };
export type { ApplicationProps };
