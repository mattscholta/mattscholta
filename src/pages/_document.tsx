import * as React from 'react';
import Document, {
  DocumentProps,
  Head,
  Html,
  Main,
  NextScript
} from 'next/document';

import { GOOGLE_TAG_MANAGER } from '@src/config/settings';
import { GoogleTagManagerIFrame } from '@src/services/tracking/components/GoogleTagManagerIFrame';
import { script as scriptArtwork } from '@src/utils/artwork';

/**
 * @name AppDocument
 * @external https://nextjs.org/docs/advanced-features/custom-document
 * @description A custom Document is commonly used to augment your
 * application's <html> and <body> tags.
 */
class AppDocument extends Document {
  render(): JSX.Element {
    return (
      <Html lang="en">
        <Head />
        <body>
          <div dangerouslySetInnerHTML={{ __html: scriptArtwork }} />
          <GoogleTagManagerIFrame id={GOOGLE_TAG_MANAGER} />
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

export { AppDocument, AppDocument as default };
export type { DocumentProps as AppDocumentProps };
