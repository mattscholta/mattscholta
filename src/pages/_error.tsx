import * as React from 'react';
import classnames from 'classnames';
import Head from 'next/head';

import { SITE_TITLE } from '@src/config/constants';

import styles from './_error.module.css';

interface ErrorPageProps {}

/**
 * @name ErrorPage
 * @description Simple 404 page
 */
const ErrorPage: React.FC<ErrorPageProps> = (_props) => {
  // Styles
  const cssComponent = classnames('ui-main', styles.component);

  return (
    <>
      <Head>
        <title>Page not found | {SITE_TITLE}</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main className={cssComponent}>
        <p>Sorry, we couldnâ€™t find the page you were looking for.</p>
      </main>
    </>
  );
};

export { ErrorPage, ErrorPage as default };
export type { ErrorPageProps };
