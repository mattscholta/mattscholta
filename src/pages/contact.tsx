import * as React from 'react';
import classnames from 'classnames';
import Head from 'next/head';

import {
  SITE_EMAIL_ADDRESS,
  SITE_EMAIL_LINK,
  SITE_TITLE
} from '@src/config/constants';

import styles from './contact.module.css';

interface ContactProps {}

/**
 * @name Contact
 * @description Application landing page (homepage)
 */
const Contact: React.FC<ContactProps> = (_props) => {
  // Styles
  const cssComponent = classnames('ui-main', styles.component);

  // Markup
  const renderedEmail = (
    <a className="ui-link" href={SITE_EMAIL_LINK}>
      {SITE_EMAIL_ADDRESS}
    </a>
  );

  return (
    <>
      <Head>
        <title>Contact | {SITE_TITLE}</title>
      </Head>

      <main className={cssComponent}>
        <div className="ui-container-xl u-p-2x">
          <h1 className={styles.h1}>Contact Info</h1>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quis saepe
            ut magni id deserunt labore dignissimos at quas asperiores ipsum ab
            quos accusamus praesentium atque magnam recusandae distinctio,
            tempore voluptatum?
          </p>
          <p>Contact Jax at {renderedEmail}...</p>
        </div>
      </main>
    </>
  );
};

export { Contact as default, Contact };
export type { ContactProps };
