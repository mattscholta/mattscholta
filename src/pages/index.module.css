.component {
  margin-top: var(--grid-5x);
}

.component aside {
  flex: 0 0 400px;
}

.component section {
  flex: 1 0 320px;
}

.component p em {
  font-weight: bold;
}

.button {
  margin: var(--grid-10x) auto var(--grid-3x);
}

.hero {
  margin-bottom: var(--grid-4x);
}

.left {
  margin-bottom: theme('spacing.8x');
}

.link {
  display: block;
  font-weight: var(--font-weight-bold);
  margin-left: var(--grid-2x);
  margin-top: var(--grid-4x);
  text-align: center;
  transition: padding 200ms ease-out;
}

.link:hover {
  padding-left: var(--grid-1x);
}

.sidebar {
  margin-bottom: var(--grid-6x);
}

@media (max-width: 1024px) {
  .component {
    margin-top: 0;
  }
}

@media (max-width: 768px) {
  .left {
    margin-bottom: theme('spacing.4x');
  }

  .link {
    margin-bottom: var(--grid-6x);
  }

  .sidebar {
    margin-bottom: var(--grid-4x);
    margin-top: var(--grid-2x);
  }
}
