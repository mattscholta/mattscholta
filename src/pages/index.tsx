import * as React from 'react';
import classnames from 'classnames';
import Head from 'next/head';
import Link from 'next/link';

import { ConnectOnline } from '@src/components/ConnectOnline';
import { DevelopmentStack } from '@src/components/DevelopmentStack';
import { WorkHistory } from '@src/components/WorkHistory';
import { CLIENT_CLOUDINARY, SITE_TITLE } from '@src/config/constants';
import { getSrcSet } from '@src/utils/links';

import styles from './index.module.css';

interface IndexProps {
  posts: any[];
}

/**
 * @name Index
 * @description Application landing page (homepage)
 */
const Index: React.FC<IndexProps> = (props) => {
  const { posts: _posts } = props;

  // Setup
  const img = `${CLIENT_CLOUDINARY}/v1616195587/mattscholta/general/coast_1500.webp`;
  const sources = getSrcSet(img);

  // Styles
  const cssComponent = classnames('ui-main', styles.component);

  return (
    <>
      <Head>
        <title>About | {SITE_TITLE}</title>
      </Head>

      <main className={cssComponent}>
        <div className="ui-container-xl u-p-2x">
          <div className="ui-reflow">
            <div className={styles.left}>
              <blockquote style={{ maxWidth: '85%' }}>
                A <em>Software Engineer</em> passionate about{' '}
                <b>quality code</b> written for <b>humans</b>, unlocking{' '}
                <b>developer productivity</b>, and creating delightful{' '}
                <b>user experiences</b>.<cite>- Matthew Scholta</cite>
              </blockquote>
              <img
                alt="The California Coast"
                className={styles.hero}
                height={500}
                loading="eager"
                sizes="(min-width: 768px) 60vw, 100vw"
                src={img}
                srcSet={sources}
                title="Life's a beach, work hard and play in the sand!"
                width={1500}
              />

              <h1 className="ui-heading">Opportunity awaits üßë‚Äçüíª</h1>
              <sub className="ui-subhead" />

              <p>
                After <b>5+ great</b> years pushing the envelope at{' '}
                <a className="u-font-bold" href="https://thredup.com">
                  thredUP.com
                </a>{' '}
                the time has come for a new set of challenges. With a focus on
                front end development, user-experience, user-interaction and a
                passion for developer productivity I am beyond excited to see
                what opportunities the future holds.
              </p>

              <p>
                The technology and tools listed on this page are my preferred
                stack as of 2022. This year I'll be launching a new version of a
                long time project{' '}
                <a className="u-font-bold" href="https://barguide.io">
                  BarGuide.io
                </a>{' '}
                with all that <b>"free time"</b> I have. ü§£
              </p>

              <p>
                So, if you're interested in learning more, ask the nearest
                developer to help you find my <b>email address</b>. I've hidden
                it into the website, twice actually... Otherwise reach out and
                connect on LinkedIn.
              </p>

              <Link href="/portfolio">
                <a className={styles.link}>View Portfolio &#8594;</a>
              </Link>
            </div>

            <div className="ui-spacer" />

            <aside className={styles.right}>
              <ConnectOnline />
              <WorkHistory />
              <DevelopmentStack />
            </aside>
          </div>
        </div>
      </main>
    </>
  );
};

export { Index as default, Index };
export type { IndexProps };
