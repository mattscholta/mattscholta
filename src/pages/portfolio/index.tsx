import * as React from 'react';
import classnames from 'classnames';
import Head from 'next/head';

import { SITE_TITLE } from '@src/config/constants';

import styles from './index.module.css';

interface IndexProps {}

/**
 * @name Index
 * @description Application landing page (homepage)
 */
const Index: React.FC<IndexProps> = (_props) => {
  // Styles
  const cssComponent = classnames('ui-main', styles.component);

  return (
    <>
      <Head>
        <title>Index | {SITE_TITLE}</title>
      </Head>

      <main className={cssComponent}>
        <div className="ui-container-xl u-p-2x">
          <h1 className={styles.h1}>Index</h1>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quis saepe
            ut magni id deserunt labore dignissimos at quas asperiores ipsum ab
            quos accusamus praesentium atque magnam recusandae distinctio,
            tempore voluptatum?
          </p>
        </div>
      </main>
    </>
  );
};

export { Index as default, Index };
export type { IndexProps };
