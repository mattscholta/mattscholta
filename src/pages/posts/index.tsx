import * as React from 'react';
import classnames from 'classnames';
import { gql } from 'graphql-request';
import Head from 'next/head';

import { BlogPreviewColumn } from '@src/components/BlogPreviewColumn';
import { SITE_TITLE } from '@src/config/constants';
import { graphcms } from '@src/utils/graphcms';

import styles from './index.module.css';

interface IndexProps {
  posts: any[];
}

/**
 * @name Index
 * @description Application landing page (homepage)
 */
const Index: React.FC<IndexProps> = (props) => {
  const { posts } = props;
  console.log(` üí¨ ~ props`, props);

  // Styles
  const cssComponent = classnames('ui-main', styles.component);
  const cssHeading = classnames('u-visually-hidden', styles.heading);

  // Markup
  const renderPreview = (post: any, index: number) => {
    const { slug } = post;
    const key = `${slug}-${index}`;

    return (
      <BlogPreviewColumn
        className={styles.preview}
        key={key}
        post={post}
        small={false}
      />
    );
  };

  return (
    <>
      <Head>
        <title>Index | {SITE_TITLE}</title>
      </Head>

      <main className={cssComponent}>
        <div className="ui-container-xl u-p-2x">
          <h1 className={cssHeading}>A Developers Ramblings</h1>

          <blockquote>
            Always leave the campground code cleaner than you found it. üèïÔ∏è
            <cite>A Developers Ramblings</cite>
          </blockquote>

          <section className="u-flex u-flex-wrap">
            {posts.map(renderPreview)}
          </section>

          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quis saepe
            ut magni id deserunt labore dignissimos at quas asperiores ipsum ab
            quos accusamus praesentium atque magnam recusandae distinctio,
            tempore voluptatum?
          </p>
        </div>
      </main>
    </>
  );
};

const QUERY = gql`
  {
    posts {
      id
      imageTemp
      title
      # content {
      #   html
      # }
    }
  }
`;

export async function getStaticProps() {
  const query = await graphcms.request(QUERY);
  const { posts } = query;

  return {
    props: {
      posts
    }
  };
}

export { Index as default, Index };
export type { IndexProps };
