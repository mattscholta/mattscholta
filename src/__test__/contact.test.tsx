import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { RecoilRoot } from 'recoil';

import { Contact, ContactProps } from '../pages/contact';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      asPath: '',
      pathname: '',
      push: jest.fn(),
      query: {},
      route: '/'
    };
  }
}));

describe('Contact', () => {
  let props: ContactProps;
  let wrapper: RenderResult;

  const createWrapper = () => (
    <RecoilRoot>
      <Contact {...props} />
    </RecoilRoot>
  );

  beforeAll(() => {
    window.IntersectionObserver = jest.fn().mockImplementation(() => ({
      disconnect: jest.fn(),
      observe: jest.fn(),
      unobserve: jest.fn()
    }));

    props = { error: false };
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
