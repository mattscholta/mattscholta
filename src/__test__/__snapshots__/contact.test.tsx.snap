// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact matches snapshot 1`] = `
NodeList [
  <main
    class="ui-main component"
  >
    <div
      class="ui-container-xl u-p-2x"
    >
      <h1
        class="h1"
      >
        Contact Info
      </h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Quis saepe ut magni id deserunt labore dignissimos at quas asperiores ipsum ab quos accusamus praesentium atque magnam recusandae distinctio, tempore voluptatum?
      </p>
      <p>
        Contact Jax at 
        <a
          class="ui-link"
          href="mailto:matthew.scholta@gmail.com"
        >
          matthew.scholta@gmail.com
        </a>
        ...
      </p>
    </div>
  </main>,
]
`;
